daffodil-core/src/main/scala/org/apache/daffodil/core/runtime1/ChoiceTermRuntime1Mixin.scala

```diff
@@ -80,36 +80,15 @@ trait ChoiceTermRuntime1Mixin { self: ChoiceTermBase =>
     // element event to be in the infoset (for round-tripping from parse). It's value will
     // get recomputed, but it can appear with a stale value (left-over from parse)
     // in the arriving infoset for unparse.
-    //
-    val optDefaultBranch = {
-      val optEmpty: Option[Term] = {
-        val emptyBranches = groupMembers.filter { gm =>
-          val ies = gm.identifyingEventsForChoiceBranch
-          ies.pnes.isEmpty // empty event list makes it the default, not simply isOpen
-        }
-        if (emptyBranches.length > 1)
-          SDW(
-            WarnID.MultipleChoiceBranches,
-            "Multiple choice branches with no required elements detected and the infoset does not specify a branch, selecting the first branch for unparsing"
-          )
-        emptyBranches.headOption
-      }
-      val optOpen: Option[Term] =
-        optEmpty.orElse {
-          groupMembers.find { gm =>
-            val ies = gm.identifyingEventsForChoiceBranch
-            ies.isOpen // first open one is used if there is no branch that has empty event list. (test ptg_1u)
-          }
-        }
-      val optDefault: Option[Term] =
-        optOpen.orElse {
-          groupMembers.find {
-            _.canUnparseIfHidden
-          } // optional, defaultable, OVC, or IVC
-        }
-      optDefault
+    // Rather than defaulting to the first empty (i.e no possible next elements) branch,
+    // we instead default to the first defaultable element (i.e has a defined dfdl:outputValueCalc,
+    // dfdl:inputValueCalc, an optional branch or an array branch with no required elements).
+    // It is important to note that canUnparseIfNoEvent subsumes empty branches, as well as
+    // open/optional branches, so we don't need to explicitly check for those
```

Comment:
I think saying "we instead default to the first defaultable element" is potentially confusing. "defaultable" could be confused with things that have `default="..."` which isn't necessary (or currently supported). And saying "element" is confusing because we could chose a branch that has no elements in it--it could just be a empty sequence with asserts, or a hidden group.

Maybe instead, just say something like

> We default to the first branch can be unparsed entirely without any infoset events (i.e. made up of elements that are dfdl:outputValueCalc, dfdl:inputValueCalc, option branches, zero length arrays, or defaultable elements)